api/views.py

```diff
@@ -139,6 +165,133 @@ def _run_job(self, job):
+    def properties(self, request, pk=None):
+        job = self.get_object()
+        properties = PrintingProperties.objects.filter(job=job).first()
+        if not properties:
+            return Response("Error: job properties not found", status=status.HTTP_404_NOT_FOUND)
+        return Response({
+            "id": job.id,
+            "status": job.status,
+            "printer": job.printer.name,
+            "copies": properties.copies,
+            "pages_to_print": properties.pages_to_print,
+            "color": properties.color,
+            "two_sides": properties.two_sides,
+            "fit_to_page": properties.fit_to_page
+        })    
+    
+    @action(detail=True, methods=['post'], name="Change properties")
+    def change_properties(self, request, pk=None):
+        job = self.get_object()
+        if job.status != JobStatus.INCOMING:
+            return Response("Error: invalid job status for this request", status=status.HTTP_400_BAD_REQUEST)
+        
+        serializer = CreatePrintJobRequestSerializer(data=request.data)
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+        
+        error = self._validate_properties(job, serializer.validated_data["printer"], PrintingProperties(
+            color=serializer.validated_data['color'],
+            copies=serializer.validated_data['copies'],
+            two_sides=serializer.validated_data['two_sides'],
+            pages_to_print=serializer.validated_data['pages_to_print'],
+            fit_to_page=serializer.validated_data['fit_to_page']
+        ))
+        if error:
+            return Response(error, status=status.HTTP_400_BAD_REQUEST)
+
+        properties = PrintingProperties.objects.get(job=job)
+        job.printer = Printer.get_printer_for_user(user=self.request.user,
+                                                   printer_id=serializer.validated_data['printer'])
+        properties.copies = serializer.validated_data['copies']
+        properties.pages_to_print = serializer.validated_data['pages_to_print']
+        properties.color = serializer.validated_data['color']
+        properties.two_sides = serializer.validated_data['two_sides']
+        properties.fit_to_page = serializer.validated_data['fit_to_page']
+        properties.save()
+
+        return Response(self.get_serializer(job).data)
+    
+    
+    @action(detail=True, methods=['get'], name='Validate job')
+    def validate_job(self, request, pk=None):
+        job = self.get_object()
+        error = self._validate_properties(job, job.printer, job.properties)
+        
+        return Response(error if error else {"error": None, "message": "Job is valid"}, 
+                        status=status.HTTP_400_BAD_REQUEST if error else status.HTTP_200_OK)
+        
+    
+    
+    def _validate_properties(self, job: GutenbergJob, printer, properties:PrintingProperties):
```

Comment:
It's not clear from the function name or signature that this method returns errors. Consider throwing them or changing the name.